mypyc/genops.py

```diff
@@ -300,24 +300,34 @@ class FuncInfo(object):
     def __init__(self,
                  callable_class: ClassIR = INVALID_CLASS,
                  env_class: ClassIR = INVALID_CLASS,
-                 env_class_val: Value = INVALID_VALUE,
                  self_reg: Value = INVALID_VALUE,
+                 env_reg: Value = INVALID_VALUE,
+                 prev_env_reg: Value = INVALID_VALUE,
                  is_nested: bool = False,
-                 contains_nested: bool = False) -> None:
+                 contains_nested: bool = False,
+                 symbol_to_env: Dict[SymbolNode, Value] = {}) -> None:
```

Comment:
Danger! This probably doesn't work right.

A quirk of python default argument values is that the same value is used every time the function is invoked. This means that every call to this that uses the default value uses the same dictionary. I think the traditional fix is to make the default value `None` and then explicitly create a new dictionary in that case.